setup-ok/action.yml

```diff
@@ -124,34 +144,70 @@ runs:
           exit 0
         fi
         
+        gh_download() {
+          local REPO="$1"
+          local VERSION="$2" 
+          local PATTERN="$3"
+          local OUTPUT="$4"
+          local DESCRIPTION="$5"
+          local MAX_RETRIES=3
+          local RETRY_COUNT=0
+
+          while [[ $RETRY_COUNT -lt $MAX_RETRIES ]]; do
+            if gh release download "$VERSION" --repo "$REPO" --pattern "$PATTERN" --output "$OUTPUT"; then
+              return 0
+            fi
+            
+            RETRY_COUNT=$((RETRY_COUNT + 1))
+            if [[ $RETRY_COUNT -lt $MAX_RETRIES ]]; then
+              echo "Failed to $DESCRIPTION. Retrying ($RETRY_COUNT/$MAX_RETRIES)..." >&2
+              sleep 2
+            fi
+          done
+          
+          echo "Error: Unable to $DESCRIPTION after $MAX_RETRIES retries." >&2
+          return 1
+        }
+
+        curl_download() {
+          local URL="$1"
+          local OUTPUT="$2"
+          local DESCRIPTION="$3"
+          local MAX_RETRIES=3
+          local RETRY_COUNT=0
+
+          while [[ $RETRY_COUNT -lt $MAX_RETRIES ]]; do
+            if curl -L "$URL" -o "$OUTPUT"; then
+              return 0
+            fi
+            
+            RETRY_COUNT=$((RETRY_COUNT + 1))
+            if [[ $RETRY_COUNT -lt $MAX_RETRIES ]]; then
+              echo "Failed to $DESCRIPTION. Retrying ($RETRY_COUNT/$MAX_RETRIES)..." >&2
+              sleep 2
+            fi
+          done
+          
+          echo "Error: Unable to $DESCRIPTION after $MAX_RETRIES retries." >&2
+          return 1
+        }
+        
         echo "Installing ok..."
-        gh release download "${{ steps.versions.outputs.ok_version }}" \
-          --repo "oslokommune/ok" \
-          --pattern "ok_*_linux_amd64.tar.gz" \
-          --output - | tar -xzOf - ok > "$BIN_DIR/ok"
+        gh_download "oslokommune/ok" "${{ steps.versions.outputs.ok_version }}" "ok_*_linux_amd64.tar.gz" "-" "install ok" | tar -xzOf - ok > "$BIN_DIR/ok"
```

Comment:
nit: Man kan redusere sannsynligheten for script injection ved å bruke f. eks. [miljøvariabler](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable) fremfor å bruke en inline expression. Vi har sannsynligvis god nok kontroll på hvordan input-variablene brukes her til at det er vits å gjøre noe med, men er ellers en grei praksis

Reply:
Bra forslag 👍 Fiksa.